### PR TITLE
adjust buffer/retry mechanism

### PR DESCRIPTION
**Pull request by wrongerror**
Signed-off-by: wrongerror <wangyifei@kubesphere.io>

### CODE REVIEW

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -33,40 +32,43 @@ type Config struct {
...
-}
-
 type Runtime struct {
 	config      *Config
 	ctx         *ofctx.FunctionContext
 	grpc        *grpc.Manager
 	funcChannel channel.AppChannel
-	reqChan     chan *EventRequest
-	respMap     *ResponseMap
+	events      chan Event
```
Comment (benjaminhuo) [inline pkg/runtime/runtime.go]: I think `events      chan *Event` will be more efficient

File: pkg/grpc/grpc.go [hunk 2]
```diff
@@ -100,17 +100,19 @@ func (g *Manager) StartEndpointsDetection() {
...
 				if err != nil {
 					klog.Error(err)
 				} else if state == connectivity.Ready || state == connectivity.Idle {
 					g.balancer.Add(ep)
+				} else {
+					g.balancer.Remove(ep)
 				}
 			}
-			time.Sleep(200 * time.Millisecond)
+			time.Sleep(time.Second)
```
Comment (benjaminhuo) [inline pkg/grpc/grpc.go]: Why changing to a longer interval?

Comment (wrongerror) [reply pkg/grpc/grpc.go:115]: Want to reduce the pressure of requesting dns

Comment (benjaminhuo) [reply pkg/grpc/grpc.go]: Are there any good effects after changes like this?

Comment (wrongerror) [reply pkg/grpc/grpc.go:115]: Apart from the reasons mentioned above, no

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**